pkg/config/tuning.go

```diff
@@ -93,6 +93,12 @@ func DefaultFairnessTrackerConfig() *FairnessTrackerConfig {
 //   - tolerableBadRequestsPerBadFlow: number of failed requests tolerated before
 //     a flow is fully blocked.
 func GenerateTunedStructureConfig(expectedClientFlows, bucketsPerLevel, tolerableBadRequestsPerBadFlow uint32) *FairnessTrackerConfig {
+	// If caller passes 0, fall back to the sane default and warn.
+    if tolerableBadRequestsPerBadFlow == 0 {
+        log.Printf("tolerableBadRequestsPerBadFlow=0 is invalid; falling back to default %d", defaultTolerableBadRequestsPerBadFlow)
+        tolerableBadRequestsPerBadFlow = defaultTolerableBadRequestsPerBadFlow
```

Comment:
I don't actually like changing this value behind user's back especially since this function is meant to be for customization. While this is part of the public API (that's sparingly used I'll add) and this will technically be a breaking change, I think in this case it's acceptable to change the signature to add error. None of the library code beyond this file depends on it. You may have to fix some tests though,